### PR TITLE
CI: Drop 2.0.0 settings, name each build job

### PR DESCRIPTION
**Pull request by olleolleolle**
TL;DR: an attempt to get to green for this repo.

- Add `name` property to matrix combinations. This allows for easier scanning of the `allow_failures` section
- Add `dist: trusty` - to allow rvm to continue past the step where gemsets failed to be created
- Changed the check-for-Rubinius versions regex to support the current output
- Changed the expectation from `2.3.1` ➡️  `10.0` ❓ ❓ ⚠️ ❓ ❓ Are there perhaps any new ways of learning the "supported version" or is `10.0` the value to check for? **Answer**: 10.0 _is_ the value to check for.

---

## Details

On a lark, I added `dist: trusty`. That made things pass far enough that Rubinius could express its `--version` output.

Then, the regex which checked it didn't like the 10.0 (expecting a triple of numbers, like `10.0.0`). I added a non-capturing regex group to also accept 10.0. `Rakefile` has the regex.

Now, it seemed that the output of `rbx-4` no longer has a value for "Ruby version compatibility" in its first line of output. Should it?

Example failure:

```
Rubinius version output:
  rubinius 4.15 (10.0 21e83c45 2020-02-02 5.0.0git-929163d) [x86_64-linux-gnu]
did not match:
  /^rubinius\s(\d+\.\d+(\.\d+)?)(\.c\d+)?\s\((\d+\.\d+\.\d+)\s[0-9a-f]{8}\s\d{4}-\d{2}-\d{2}\s\d+\.\d+(\.\d+)?(?:git\-[0-9a-f]{7})?(\sJ?I?C?D?)?\)\s\[[^\]]+\]$/
```

### CODE REVIEW

**Comment by brixen** [comment]
@olleolleolle thanks for working on this! The Ruby compatibility version is 10.0. I know Ruby likely won't reach that version this century, but Rubinius is kind of from the future. 😄 

Thoughts on what to do about the RVM gemset issue? RVM issues have been a problem for a long time but I'm out of ideas for addressing them.

**Comment by olleolleolle** [comment]
I don't know why the gemset _name_ is the empty string.

Could that be a lead?
